Choose your own adventure MVP/React skeleton

maoivy opened the pull request:
Implemented basic choices and choice tracking in the React skeleton (see #56 for the React skeleton, #45 for the meta issue). Most direct next steps would be integrating with backend to get the adventure JSON objects

ryaanahmed commented on backend/config/urls.py:
```diff
@@ -55,6 +55,7 @@ def edx_path(route, component_name):
     # React views
     path('', render_react_view, {'component_name': 'IndexView'}),
     path('all_view/', render_react_view, {'component_name': 'AllView'}),
+    path('adventure/', render_react_view, {'component_name': 'ChooseAdventureView'}),
```
use the `edx_path` helper function here, so that your thing shows up in fake edxview

ryaanahmed commented on frontend/src/chooseAdventure/chooseAdventure.js:
```diff
@@ -0,0 +1,84 @@
+
+    updateSuccess = (successFactor) => {
+        this.setState((prevState) => ({
+            successTotal: prevState.successTotal + successFactor,
```
This being additive feels weird to me -- I was expecting each step to have a likelihood of success, and that you'd get some kind of compounding probability thing at the end. When I play this otherwise the "success score" doesn't feel meaningful -- whereas some kind of probability would.

ryaanahmed commented on frontend/src/chooseAdventure/introView.js:
```diff
@@ -0,0 +1,44 @@
+                    Get started
+                </button>
+            </div>
+
```
i know this is just mvp / prototyping code but whoa there unnecessary whitespace

ryaanahmed commented on frontend/src/chooseAdventure/stageView.js:
```diff
@@ -0,0 +1,130 @@
+    setStage: PropTypes.func,
+};
+
+/**
```
nice work with this wrapper component!

mscuthbert commented:
(note also the conflicting stylesheet that needs to be fixed before merge)

ryaanahmed commented:
@Dylan-Weber @maoivy @Mayowa99 -- if someone could turnaround the smaller changes listed here sooner-rather-than-later, then we can merge, deploy, and have @evanlieb give it a spin before our Friday sessions. That'll put is in a strong position to make changes moving into the final version.

maoivy commented:
@ryaanahmed could you take a look to see if I've addressed everything correctly?

Changes in the latest commit:
- changes the view to edX view
- adds a successDetail property to each option that goes into further detail about how choosing that option might affect your chance of success 
   -  e.g., if you choose Facebook but Facebook is banned in the country, it says so.
   - should this be displayed as you go along, or only at the end?
- changed success calculation to be a multiplier: player starts out with 1
   - e.g., if a (general) media campaign has a 90% chance of working and then choosing Twitter has a 60% chance of working, the player ends with a 54% (or 0.54) chance of succeeding
   - should probably refine the calculation

ryaanahmed commented:
@maoivy -- yes! This is fantastic. Just gave it a whirl, and I can already feel how this will be a useful teaching tool.

> should this be displayed as you go along, or only at the end?
I think only at the end fits Evan's idea that the player first does things based on their own assumptions and has the underlying issues revealed at the end.
